lib/srv/authhandlers.go

```diff
@@ -460,13 +450,3 @@ func (h *AuthHandlers) authorityForCert(caType services.CertAuthType, key ssh.Pu
 func (h *AuthHandlers) isProxy() bool {
 	return h.Component == teleport.ComponentProxy
 }
-
-// extractRolesFromCert extracts roles from certificate metadata extensions.
-func extractRolesFromCert(cert *ssh.Certificate) ([]string, error) {
-	data, ok := cert.Extensions[teleport.CertExtensionTeleportRoles]
-	if !ok {
-		// it's ok to not have any roles in the metadata
-		return nil, nil
-	}
-	return services.UnmarshalCertRoles(data)
-}
```

Comment:
I think this was for old style SSH certificates, do you know if we've dropped support for them?

Reply:
@russjones This logic is still being used - this was just a duplicate function. [ExtractFromCertificate](https://github.com/gravitational/teleport/blob/v7.0.0-dev.9/lib/services/role.go#L894-L920) does pretty much the same as far as I can tell, unless I'm missing something,